### PR TITLE
Add native tls support for actix_tls::connect module

### PR DESCRIPTION
**Pull request by fakeshadow**
## PR Type

Feature

## PR Checklist
Check your PR fulfills the following:

- [ ] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

native tls was lacking on client connect module while impled in server accept module.
This PR add support for it in client connect module and can be enabled with `native-tls` feature.

No test was added for this feature as current  tests does not actually test any tls connectors(no matter openssl, rustls or native-tls). So there is no point adding test for it until this issue sorted out.

Some clippy warning fix for nightly has also been worked into this PR.

### CODE REVIEW

File: actix-tls/src/connect/ssl/mod.rs [hunk 1]
```diff
@@ -5,3 +5,6 @@ pub mod openssl;
 
 #[cfg(feature = "rustls")]
 pub mod rustls;
+
+#[cfg(feature = "native-tls")]
+pub mod nativetls;
```
Comment (robjtede) [inline actix-tls/src/connect/ssl/mod.rs:10]: nit but the mod name should match the crate name like `native_tls`

Comment (fakeshadow) [reply actix-tls/src/connect/ssl/mod.rs:10]: yea. I was match the naming with accept module. Should this be a break change PR and re-name that too?

Comment (robjtede) [reply actix-tls/src/connect/ssl/mod.rs:10]: sure

**Review by robjtede — APPROVED** [review]

**Comment by robjtede** [comment]
theres a conflict i can't solve in the UI apparently

**Pull request merged**